lib/querystring.js

```diff
@@ -200,14 +203,34 @@ function stringify(obj, sep, eq, options) {
 function charCodes(str) {
   if (str.length === 0) return [];
   if (str.length === 1) return [str.charCodeAt(0)];
-  const ret = [];
+  const ret = new Array(str.length);
   for (var i = 0; i < str.length; ++i)
-    ret[ret.length] = str.charCodeAt(i);
+    ret[i] = str.charCodeAt(i);
```

Comment:
Perhaps the most performant approach would be to use the push() method:
```
  const ret = [];
  for (var i = 0; i < str.length; ++i)
    ret.push(str.charCodeAt(i));
```
v8 team recommends this in the most cases: https://v8.dev/blog/elements-kinds#avoid-creating-holes